### ISSUE TITLE
Avoid loading all timestamps into memory

### ISSUE THREAD

**Issue by rly** [LorenFrankLab/spikegadgets_to_nwb#47]
Becomes a problem when the recording is 17 hours long. Explore whether using a DataChunkIterator is possible.

**Comment by edeno** [comment]
## Investigation (2026-06-11): measured, scoped, and confirmed unsolved

### Measured footprint
Timestamps are computed in `RecFileDataChunkIterator` ([`convert_ephys.py:206`](https://github.com/LorenFrankLab/trodes_to_nwb/blob/main/src/trodes_to_nwb/convert_ephys.py#L206)) as a single concatenated **`float64`** array (8 bytes/sample). Measured from `get_regressed_systime` on the sample rec (2.51 MB / 314k samples) and extrapolated to 30 kHz:

| Duration | timestamps array | + `np.diff` transient = peak |
|---|---|---|
| 1 h | 0.9 GB | 1.7 GB |
| **17 h** (the reported case) | **14.7 GB** | **~29 GB** |
| 39.77 h (uint32 wrap limit) | 34.4 GB | ~69 GB |

The `is_timestamps_sequential = np.all(np.diff(self.timestamps))` check ([`convert_ephys.py:219`](https://github.com/LorenFrankLab/trodes_to_nwb/blob/main/src/trodes_to_nwb/convert_ephys.py#L219)) allocates a **second** full array, so peak is ~2x the array. The ephys *data* is already chunked via the DataChunkIterator; only the **timestamps** are fully materialized — this is the bug.

### The array is reused by four consumers (not equally lazy-friendly)
| Consumer | Use | Lazy-friendly? |
|---|---|---|
| `convert_ephys` ElectricalSeries (`:410`) | sequential write | ✅ wrap timestamps in a `DataChunkIterator` |
| `convert_analog` TimeSeries (`:99`) | same-rate, sequential | ✅ **link** to the ephys timestamps |
| `convert_intervals` sample_count (`:94`) | `np.array(...)` **copy** + full `trodes_sample` | ✅ DataChunkIterators (today adds *another* ~15 GB copy) |
| `convert_position` (`:731`) | `np.digitize(epoch_interval, timestamps)` + slice | ❌ **needs random access — the blocker** |

The position path binary-searches the full array to map each epochs camera interval to sample indices. That defeats naive lazy loading, but is solvable: timestamps are a **linear regression** of the sample counter, so epoch-boundary indices can be computed analytically (or via a small per-file boundary index) instead of `np.digitize` over ~1.8 B samples.

### Not solved by any open PR; overlaps must be sequenced
Diffed the timestamp-touching PRs: **#180** changes `get_regressed_systime` internals only; **#181** rewrites the monotonicity line but still allocates the full array. So neither addresses #47 — and a fix **overlaps #181 (the `:219` monotonicity line) and interacts with #180 (`get_regressed_systime`)**. Recommend implementing on its own branch off `main`, **after #180/#181 merge** (or rebased onto them).

### Proposed approach (deferred)
- **Full fix:** `DataChunkIterator` timestamps for ephys/intervals, link analog → ephys, stream the monotonicity check (no full `np.diff`), refactor position to derive epoch-boundary indices analytically. Cuts the 17 h peak from ~29-45 GB to ~MBs.
- **Partial win (if a smaller change is preferred):** stream the monotonicity check, drop the `convert_intervals` `np.array` copy, and link analog → ephys — removes the duplicates/transients (~45 GB → ~15 GB) while leaving positions one array in place.

Deferring implementation for now per maintainer direction.
